src/main/scala/org/allenai/pnp/semparse/EntityLinking.scala

```diff
@@ -75,4 +77,13 @@ case class Span(val start: Int, val end: Int) {
   def contains(index: Int): Boolean = {
     index >= start && index < end
   }
+}
+
+case class KnowledgeGraph(val nodePairs: List[(String, String)]) {
+  val nodeSet = nodePairs.flatMap(p => List(p._1, p._2)).toSet
```

Comment:
Ah, yeah, sorry, I was just confused by your `.flatMap`.  You want `nodePairs.flatten.toSet`.  Oh, except you can't, because they're tuples...  I get it now.